src/main/java/edu/oregonstate/cs361/battleship/Main.java

```diff
@@ -182,137 +143,114 @@ public static String placeShipUpdated(Request req) {
         Point computerEnd;
+        Point cord = new Point(0,0);
+
         int computer_x = 0;
         int computer_y = 0;
         int horizontal = 0;
-        int computer_length_increase = 0;
-        int computerLength = model.getShipByID(id).getLength() - 1;
 
-        if (id.toLowerCase().contains("aircraftcarrier")) {
-            CAircraftCarrier.setStart(0, 0);
-            CAircraftCarrier.setEnd(0, 0);
-            CAircraftCarrier.setHealth(5);
+        for(int id = 0; id < 5; id++) {
+            int stop = 0;
 
-        } else if (id.toLowerCase().contains("battleship")) {
-            CBattleship.setStart(0, 0);
-            CBattleship.setEnd(0, 0);
-            CBattleship.setHealth(4);
+            int computer_length_increase = 0;
+            int computerLength = model.getShipByID(CArray[id].getName()).getLength() - 1;
 
-        } else if (id.toLowerCase().contains("submarine")) {
-            CSubmarine.setStart(0, 0);
-            CSubmarine.setEnd(0, 0);
-            CSubmarine.setHealth(3);
+            model.getShipByID(CArray[id].getName()).setStart(0, 0);
+            model.getShipByID(CArray[id].getName()).setEnd(0, 0);
+            CArray = model.resetArrayUpdated(model, false);
 
-        } else if (id.toLowerCase().contains("clipper")) {
-            CClipper.setStart(0, 0);
-            CClipper.setEnd(0, 0);
-            CClipper.setHealth(1);
+            int movingPoint = 0, stoppedPoint = 0;
 
-        } else if (id.toLowerCase().contains("dinghy")) {
-            CDinghy.setStart(0, 0);
-            CDinghy.setEnd(0, 0);
-            CDinghy.setHealth(1);
+            while (stop == 0) {
+                computer_x = (int) (Math.random() * 10 + 1);
+                computer_y = (int) (Math.random() * 10 + 1);
+                horizontal = (int) (Math.random() * 2 + 1);
 
-        }
+                computerStart = new Point(computer_x, computer_y);
+                computerEnd = new Point(0, 0);
+                stop = 1;
 
-        int movingPoint = 0, stoppedPoint = 0;
+                if (horizontal == 1 && computer_x + computerLength < 11) { //horizontal
+                    computerEnd = new Point(computer_x + computerLength, computer_y);
+                    computer_length_increase = computerEnd.getAcross();
 
-        while (stop == 0) {
-            computer_x = (int) (Math.random() * 10 + 1);
-            computer_y = (int) (Math.random() * 10 + 1);
-            horizontal = (int) (Math.random() * 2 + 1);
+                    movingPoint = computer_x;
+                    stoppedPoint = computer_y;
 
-            computerStart = new Point(computer_x, computer_y);
-            computerEnd = new Point(0, 0);
-            stop = 1;
+                } else if (horizontal == 2 && computer_y + computerLength < 11) { //vertical
+                    computerEnd = new Point(computer_x, computer_y + computerLength);
+                    computer_length_increase = computerEnd.getDown();
```

Comment:
Yeah the logic randomizes orientation just fine, just not a number I'm used to seeing in a true/false situation. No need to change.